### PR TITLE
Convert some command calls to the API

### PR DESCRIPTION
**Pull request by XVilka**
# DO NOT SQUASH

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- Removed dead code from `c` handler
- Converted `wao` command call to the API call
- Converted `e bin.str.purge` call to the API call

**Test plan**

CI is green.

### CODE REVIEW

File: librz/core/core.c [hunk 1]
```diff
@@ -3529,3 +3529,23 @@ RZ_API RzCmdStatus rz_core_core_plugins_print(RzCore *core, RzCmdStateOutput *st
 	rz_cmd_state_output_array_end(state);
 	return RZ_CMD_STATUS_OK;
 }
+
+/* \brief Seek to the last instruction in the current basic block */
+RZ_IPI bool rz_core_seek_bb_instruction(RzCore *core, int index) {
```
Comment (ret2libc) [inline librz/core/core.c]: move it to seek.c

File: librz/core/cmd/cmd.c [hunk 2]
```diff
@@ -1995,22 +1995,8 @@ static int rz_core_cmd_subst_i(RzCore *core, char *cmd, char *colon, bool *tmpse
 			case 'B': // "@B:#" // seek to the last instruction in current bb
 			{
 				int index = (int)rz_num_math(core->num, ptr + 2);
-				RzAnalysisBlock *bb = rz_analysis_find_most_relevant_block_in(core->analysis, core->offset);
```
Comment (ret2libc) [inline librz/core/cmd/cmd.c:1998]: replace also code in `tmp_nthi_stmt`.

File: librz/core/core.c [hunk 3]
```diff
@@ -3529,3 +3529,23 @@ RZ_API RzCmdStatus rz_core_core_plugins_print(RzCore *core, RzCmdStateOutput *st
 	rz_cmd_state_output_array_end(state);
 	return RZ_CMD_STATUS_OK;
 }
+
+/* \brief Seek to the last instruction in the current basic block */
```
Comment (ret2libc) [inline librz/core/core.c]: it's not just the last actually.

File: librz/core/seek.c [hunk 4]
```diff
@@ -444,3 +444,23 @@ RZ_API RzList *rz_core_seek_list(RzCore *core) {
...
+/* \brief Seek to the \p index instruction in the current basic block */
+RZ_IPI bool rz_core_seek_bb_instruction(RzCore *core, int index) {
+	RzAnalysisBlock *bb = rz_analysis_find_most_relevant_block_in(core->analysis, core->offset);
+	if (!bb) {
+		RZ_LOG_ERROR("Can't find a basic block for 0x%08" PFMT64x "\n", core->offset);
+		return false;
+	}
+	// handle negative indices
+	if (index < 0) {
+		index = bb->ninstr + index;
```
Comment (p371jah) [inline librz/core/seek.c:461]: Document this behaviour.

Comment (XVilka) [reply librz/core/seek.c:461]: Good catch, added in the Doxygen description.

File: librz/core/cmd/cmd.c [hunk 5]
```diff
@@ -4112,21 +4098,8 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(tmp_nthi_stmt) {
...
-		if (index >= 0 && index < bb->ninstr) {
-			ut16 inst_off = rz_analysis_block_get_op_offset(bb, index);
-			rz_core_seek(core, bb->addr + inst_off, true);
-		} else {
-			eprintf("The current basic block has just %d instructions\n", bb->ninstr);
-		}
-	} else {
-		eprintf("Can't find a basic block for 0x%08" PFMT64x "\n", core->offset);
+	if (rz_core_seek_bb_instruction(core, index)) {
+		core->tmpseek = true;
```
Comment (ret2libc) [inline librz/core/cmd/cmd.c:4102]: :( only cmd_search remains that use this thing...

File: librz/core/seek.c [hunk 6]
```diff
@@ -444,3 +444,27 @@ RZ_API RzList *rz_core_seek_list(RzCore *core) {
...
+	}
+	if (!(index >= 0 && index < bb->ninstr)) {
+		RZ_LOG_ERROR("The current basic block has %d instructions\n", bb->ninstr);
+		return false;
+	}
+	ut64 inst_addr = rz_analysis_block_get_op_addr(bb, index);
+	rz_core_seek(core, inst_addr, true);
+	return true;
```
Comment (ret2libc) [inline librz/core/seek.c]:
```suggestion
	return rz_core_seek(core, inst_addr, true);
```

**Review by ret2libc — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Pull request merged**